lib/Screens/sign_up_screen.dart

```diff
@@ -0,0 +1,122 @@
+    Navigator.pushReplacement(
+      context,
+      MaterialPageRoute(builder: (context) => const HomePageScreen()),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text('login screen'),
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          // ignore: prefer_const_literals_to_create_immutables
+          children: [
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 30.0),
+              child: Form(
+                  child: Column(
+                children: [
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 15),
+                    child: TextFormField(
+                      controller: userNameController,
+                      keyboardType: TextInputType.name,
+                      decoration: const InputDecoration(
+                        labelText: "username",
+                        hintText: "enter username",
+                        prefixIcon: Icon(Icons.person),
+                        border: OutlineInputBorder(),
+                      ),
+                    ),
+                  ),
+                  const SizedBox(height: 30),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 15),
+                    child: TextFormField(
+                      controller: passwordController,
+                      keyboardType: TextInputType.visiblePassword,
+                      decoration: const InputDecoration(
+                        labelText: "password",
+                        hintText: "enter password",
+                        prefixIcon: Icon(Icons.password),
+                        border: OutlineInputBorder(),
+                      ),
+                    ),
+                  ),
+                  const SizedBox(height: 30),
+                  GestureDetector(
+                    child: Padding(
+                      padding: const EdgeInsets.symmetric(horizontal: 35),
+                      child: MaterialButton(
+                        minWidth: double.infinity,
+                        onPressed: () async {
+                          String username = userNameController.text.toString();
+                          String password = passwordController.text.toString();
+                          if (username.isNotEmpty && password.isNotEmpty) {
+                            Future.delayed(Duration.zero).then((value) =>
+                                showSnackBar(context, "logging in...."));
+                            bool checkLogin =
+                                await Authentication.login(username, password);
+                            if (checkLogin) {
+                              await Authentication.getUser();
+                              await Authentication.getAllUsers();
```

Comment:
Getting logged in user information immediately after login makes sense. However, fetching all users should be done in the appropriate screen. 
If the application displays many information in the home page, you would not fetch everything in login screen then navigate. You would actually login quickly, navigate to home page and then fetch all information so the user sees that something is loading in the main screen. 

I strongly suggest to move this code `await Authentication.getAllUsers();` to `users_screen.dart`